### PR TITLE
Updated plugin to target sdk 33

### LINKED ISSUE #34: java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.Long
Getting this exception while passing the trackingInterval as integer

Can you pls update this lib based on the solution provided in the below link

https://github.com/dotintent/FlutterBleLib/commit/4069052715e5e9e17c831f597f55fdeb62548f4c

### LINKED ISSUE #36: build failed with and exception
Apologies for a newbie question, I just created a new project in flutter , added the dependencies and the example code provide and ran, It gives me the following errors.
How do I solve this ?
```

Parameter format not correct -
e: C:\flutter_windows_2.5.3-stable\flutter\.pub-cache\hosted\pub.dartlang.org\background_location_tracker-1.2.0\android\src\main\kotlin\com\icapps\background_location_tracker\BackgroundLocationTrackerPlugin.kt: (114, 13): Class 'ProxyLifecycleProvider' is not abstract and does not implement abstract member public abstract fun onActivitySaveInstanceState(@NonNull p0: Activity, @NonNull p1: Bundle): Unit defined in android.app.Application.ActivityLifecycleCallbacks
e: C:\flutter_windows_2.5.3-stable\flutter\.pub-cache\hosted\pub.dartlang.org\background_location_tracker-1.2.0\android\src\main\kotlin\com\icapps\background_location_tracker\BackgroundLocationTrackerPlugin.kt: (157, 9): 'onActivitySaveInstanceState' overrides nothing
e: C:\flutter_windows_2.5.3-stable\flutter\.pub-cache\hosted\pub.dartlang.org\background_location_tracker-1.2.0\android\src\main\kotlin\com\icapps\background_location_tracker\utils\ActivityCounter.kt: (8, 1): Object 'ActivityCounter' is not abstract and does not implement abstract member public abstract fun onActivitySaveInstanceState(@NonNull p0: Activity, @NonNull p1: Bundle): Unit defined in android.app.Application.ActivityLifecycleCallbacks
e: C:\flutter_windows_2.5.3-stable\flutter\.pub-cache\hosted\pub.dartlang.org\background_location_tracker-1.2.0\android\src\main\kotlin\com\icapps\background_location_tracker\utils\ActivityCounter.kt: (35, 5): 'onActivitySaveInstanceState' overrides nothing

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':background_location_tracker:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
BUILD FAILED in 19s
Exception: Gradle task assembleDebug failed with exit code 1
Exited (sigterm)
```

### PR DESCRIPTION
**Pull request by vanlooverenkoen**
Fixed #34 
Fixed #36

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,8 +74,42 @@ import background_location_tracker
 }
 ```
 
+## Dart implementation
```
Comment (NicolaVerbeeck) [inline README.md:77]: Flutter implementation

File: android/build.gradle [hunk 2]
```diff
@@ -2,30 +2,30 @@ group 'com.icapps.background_location_tracker'
 version '1.0-SNAPSHOT'
 
 buildscript {
-    ext.kotlin_version = '1.3.50'
+    ext.kotlin_version = '1.6.0'
```
Comment (NicolaVerbeeck) [inline android/build.gradle:5]: Kotlin is now at 1.7

File: android/src/main/kotlin/com/icapps/background_location_tracker/flutter/FlutterBackgroundManager.kt [hunk 3]
```diff
@@ -2,25 +2,28 @@ package com.icapps.background_location_tracker.flutter
...
     private const val BACKGROUND_CHANNEL_NAME = "com.icapps.background_location_tracker/background_channel"
 
+    private val flutterLoader = FlutterLoader()
+
     private fun getInitializedFlutterEngine(ctx: Context): FlutterEngine {
         Logger.debug("BackgroundManager", "Creating new engine")
 
         val engine = FlutterEngine(ctx)
-        FlutterMain.ensureInitializationComplete(ctx, null)
         //Backwards compatibility with v1. We register all the user's plugins.
```
Comment (NicolaVerbeeck) [inline android/src/main/kotlin/com/icapps/background_location_tracker/flutter/FlutterBackgroundManager.kt:27]: Should we still retain this compatibility?

File: android/src/main/kotlin/com/icapps/background_location_tracker/service/LocationUpdatesService.kt [hunk 4]
```diff
@@ -100,17 +101,26 @@ internal class LocationUpdatesService : Service() {
...
         changingConfiguration = false
         return binder
     }
 
+    @Suppress("DEPRECATION")
+    private fun stopForeground() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            stopForeground(STOP_FOREGROUND_REMOVE)
+        } else {
+            stopForeground(true)
```
Comment (NicolaVerbeeck) [inline android/src/main/kotlin/com/icapps/background_location_tracker/service/LocationUpdatesService.kt:114]: suppress the deprecation on this statement instead of the method, this allows us to catch future deprecations

File: android/src/main/kotlin/com/icapps/background_location_tracker/service/LocationUpdatesService.kt [hunk 5]
```diff
@@ -100,17 +101,26 @@ internal class LocationUpdatesService : Service() {
...
+        }
+    }
+
     override fun onRebind(intent: Intent) {
         // Called when a client (MainActivity in case of this sample) returns to the foreground
         // and binds once again with this service. The service should cease to be a foreground
         // service when that happens.
         Logger.debug(TAG, "OnRebind")
-        stopForeground(true)
+        stopForeground()
```
Comment (NicolaVerbeeck) [inline android/src/main/kotlin/com/icapps/background_location_tracker/service/LocationUpdatesService.kt:123]: Perhaps we should try to avoid naming our methods the same as the framework methods

File: android/src/main/kotlin/com/icapps/background_location_tracker/service/LocationUpdatesService.kt [hunk 6]
```diff
@@ -153,6 +163,8 @@ internal class LocationUpdatesService : Service() {
         } else {
             startService(Intent(applicationContext, LocationUpdatesService::class.java))
         }
+        val locationRequest = this.locationRequest ?: return
```
Comment (NicolaVerbeeck) [inline android/src/main/kotlin/com/icapps/background_location_tracker/service/LocationUpdatesService.kt:166]: I don't think this is required here

File: android/src/main/kotlin/com/icapps/background_location_tracker/service/LocationUpdatesService.kt [hunk 7]
```diff
@@ -173,6 +185,7 @@ internal class LocationUpdatesService : Service() {
             wakeLock?.release();
         }
         Logger.debug(TAG, "Removing location updates")
+        val locationCallback = this.locationCallback ?: return
```
Comment (NicolaVerbeeck) [inline android/src/main/kotlin/com/icapps/background_location_tracker/service/LocationUpdatesService.kt:188]: Same for this

File: android/src/main/kotlin/com/icapps/background_location_tracker/utils/NotificationUtil.kt [hunk 8]
```diff
@@ -47,9 +49,17 @@ internal object NotificationUtil {
     private fun getNotification(context: Context, location: Location?): Notification {
         val intent = Intent(context, LocationUpdatesService::class.java)
         intent.putExtra(LocationUpdatesService.EXTRA_STARTED_FROM_NOTIFICATION, true)
-        val cancelTrackingIntent = PendingIntent.getService(context, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT)
+        val cancelTrackingIntent = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            PendingIntent.getService(context, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE)
```
Comment (NicolaVerbeeck) [inline android/src/main/kotlin/com/icapps/background_location_tracker/utils/NotificationUtil.kt:53]: This seems conflicting, update current and immutable at the same time?

File: pubspec.yaml [hunk 9]
```diff
@@ -10,6 +10,7 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
+  shared_preferences: ^2.0.15
```
Comment (NicolaVerbeeck) [inline pubspec.yaml:13]: Why was this added? Can't seem to find references in the changed code for this

**Review by NicolaVerbeeck — CHANGES_REQUESTED** [review]

File: android/src/main/kotlin/com/icapps/background_location_tracker/flutter/FlutterBackgroundManager.kt (hunk 3 above)
Comment (vanlooverenkoen) [reply android/src/main/kotlin/com/icapps/background_location_tracker/flutter/FlutterBackgroundManager.kt:27]: Other plugins still have this support. So I would still keep this. (If not remove it in another pr with another build)

File: android/src/main/kotlin/com/icapps/background_location_tracker/service/LocationUpdatesService.kt (hunk 6 above)
Comment (vanlooverenkoen) [reply android/src/main/kotlin/com/icapps/background_location_tracker/service/LocationUpdatesService.kt:166]: requestLocationUpdates requires a non nullable value

File: android/src/main/kotlin/com/icapps/background_location_tracker/service/LocationUpdatesService.kt (hunk 7 above)
Comment (vanlooverenkoen) [reply android/src/main/kotlin/com/icapps/background_location_tracker/service/LocationUpdatesService.kt:188]: removeLocationUpdates requires a non nullable value

File: pubspec.yaml (hunk 9 above)
Comment (vanlooverenkoen) [reply pubspec.yaml]: Oh, this should only be added in the example project. my bad

**Review by NicolaVerbeeck — APPROVED** [review]

**Comment by NicolaVerbeeck** [comment]
LGTM
